packages/medusa/src/api/routes/admin/discounts/delete-condition.ts

```diff
@@ -0,0 +1,84 @@
+import { IsOptional, IsString } from "class-validator"
+import { defaultAdminDiscountsFields, defaultAdminDiscountsRelations } from "."
+import { Discount } from "../../../../models"
+import { DiscountService } from "../../../../services"
+import DiscountConditionService from "../../../../services/discount-condition"
+import { getRetrieveConfig } from "../../../../utils/get-query-config"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [delete] /discounts/{id}/conditions/{condition_id}
+ * operationId: "DeleteDiscountsDiscountCondition"
+ * summary: "Delete a DiscountCondition"
+ * description: "Deletes a DiscountCondition"
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The id of the Discount
+ *   - (path) condition_id=* {string} The id of the DiscountCondition
+ * tags:
+ *   - Discount
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             id:
+ *               type: string
+ *               description: The id of the deleted DiscountCondition
+ *             object:
+ *               type: string
+ *               description: The type of the object that was deleted.
+ *             deleted:
+ *               type: boolean
+ *             discount:
+ *               type: object
+ *               description: The Discount to which the condition used to belong
+ */
+export default async (req, res) => {
+  const { discount_id, condition_id } = req.params
+
+  const validatedParams = await validator(
+    AdminDeleteDiscountsConditionsConditionParams,
+    req.query
+  )
+
+  const conditionService: DiscountConditionService = req.scope.resolve(
+    "discountConditionService"
+  )
+  const discountService: DiscountService = req.scope.resolve("discountService")
+
+  // ensure that we delete condition from existing discount
+  let discount = await discountService.retrieve(discount_id, {
```

Comment:
@olivermrbl we are retrieving the discount here, that's nice, but we still don't validate that the condition is part of that discount.

I would suggest to left join the condition here so that if the condition exists we get either the condition we are looking for and we can remove it, or the condition is not present and we can return early. wdyt?

Reply:
Also, since we do not use that discount and it is just a guard, I suggest that we do not store the result and instead the discount retrieval at the end become a `const`